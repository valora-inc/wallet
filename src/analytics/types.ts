export enum ScrollDirection {
  next = 'next',
  previous = 'previous',
}

export enum BackQuizProgress {
  word_chosen = 'word_chosen',
  backspace = 'backspace',
}

// Origin of Send actions
export enum SendOrigin {
  AppSendFlow = 'app_send_flow', // Sending as part of the app send flow
  AppRequestFlow = 'app_request_flow', // Sending because of a received payment request
  Bidali = 'bidali', // Sending from Bidali
}

// Origin of WalletConnect pairing
export enum WalletConnectPairingOrigin {
  Scan = 'scan',
  Deeplink = 'deeplink',
}

// Origin of WalletConnect/DappKit request
export enum DappRequestOrigin {
  InAppWebView = 'in_app_web_view',
  External = 'external',
}
