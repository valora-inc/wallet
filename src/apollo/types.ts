/*
 * This file was originally generated by `yarn run build:gen-graphql-types`, but
 * has since been manually edited and graphql codegen has been removed.
 * TODO(any): cleanup unused types from this file
 */

import BigNumber from 'bignumber.js'

export type Maybe<T> = T | null

/** All built-in and custom scalars, mapped to their actual values */
export interface Scalars {
  ID: string
  String: string
  Boolean: boolean
  Int: number
  Float: number
  Address: string
  Timestamp: number
  Decimal: BigNumber.Value
  Upload: any
}

export interface LocalMoneyAmount {
  __typename?: 'LocalMoneyAmount'
  value: Scalars['Decimal']
  currencyCode: Scalars['String']
  exchangeRate: Scalars['Decimal']
}

export interface MoneyAmount {
  __typename?: 'MoneyAmount'
  value: Scalars['Decimal']
  currencyCode: Scalars['String']
  localAmount?: Maybe<LocalMoneyAmount>
}

export interface ExchangeRateQueryVariables {
  currencyCode: Scalars['String']
  sourceCurrencyCode?: Maybe<Scalars['String']>
}

export interface ExchangeRateQuery {
  __typename?: 'Query'
  currencyConversion: Maybe<{ __typename?: 'ExchangeRate'; rate: BigNumber.Value }>
}

export interface IntrospectionResultData {
  __schema: {
    types: Array<{
      kind: string
      name: string
      possibleTypes: Array<{
        name: string
      }>
    }>
  }
}
export const introspectionQueryResultData: IntrospectionResultData = {
  __schema: {
    types: [
      {
        kind: 'UNION',
        name: 'Event',
        possibleTypes: [
          {
            name: 'Transfer',
          },
        ],
      },
      {
        kind: 'INTERFACE',
        name: 'TokenTransaction',
        possibleTypes: [
          {
            name: 'TokenTransfer',
          },
        ],
      },
    ],
  },
}
