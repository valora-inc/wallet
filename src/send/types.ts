export enum InviteRewardsType {
  NFT = 'nft',
  CUSD = 'cUSD',
  NONE = 'none',
}
