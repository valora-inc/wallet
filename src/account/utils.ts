import { parsePhoneNumber } from '@celo/phone-utils'

const ADDRESS_LENGTH = 42 // TODO(ACT-1173): see if this can be replaced with a viem helper

export const isAddressFormat = (content: string): boolean => {
  return content.startsWith('0x') && content.length === ADDRESS_LENGTH
}

export function getPhoneNumberDetails(
  phoneNumber: string,
  countryCallingCode: string,
  countryCodeAlpha2: string
) {
  const phoneDetails = parsePhoneNumber(phoneNumber, countryCallingCode)

  if (phoneDetails) {
    return {
      // Show international display number to avoid confusion
      internationalPhoneNumber: phoneDetails.displayNumberInternational,
      e164Number: phoneDetails.e164Number,
      isValidNumber: true,
      countryCodeAlpha2: phoneDetails.regionCode!,
    }
  } else {
    return {
      internationalPhoneNumber: phoneNumber,
      e164Number: '',
      isValidNumber: false,
      countryCodeAlpha2,
    }
  }
}
