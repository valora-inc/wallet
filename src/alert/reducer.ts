import { Action } from '@reduxjs/toolkit'
import { Actions, ActionTypes, AlertTypes } from 'src/alert/actions'
import { ErrorMessages } from 'src/app/ErrorMessages'
import { RootState } from 'src/redux/reducers'

export enum ErrorDisplayType {
  'BANNER',
  'INLINE',
}

export interface Alert {
  type: AlertTypes
  displayMethod: ErrorDisplayType
  message: string
  dismissAfter?: number | null
  buttonMessage?: string | null
  action?: Action | null
  title?: string | null
  underlyingError?: ErrorMessages | null
}

type State = Alert | null

const initialState = null

export const reducer = (state: State = initialState, action: ActionTypes): State => {
  switch (action.type) {
    case Actions.SHOW:
      return {
        displayMethod: action.displayMethod,
        type: action.alertType,
        message: action.message,
        dismissAfter: action.dismissAfter,
        buttonMessage: action.buttonMessage,
        action: action.action,
        title: action.title,
        underlyingError: action.underlyingError,
      }
    case Actions.HIDE:
      return null
    default:
      if (state?.action === action) {
        // Hide alert when the alert action is dispatched
        return null
      }
      return state
  }
}

export const errorSelector = (state: RootState) => {
  return state.alert ? state.alert.underlyingError || null : null
}
