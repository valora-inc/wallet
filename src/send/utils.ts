import { SendOrigin } from 'src/analytics/types'
import { LocalCurrencyCode } from 'src/localCurrency/consts'
import {
  convertDollarsToLocalAmount,
  convertLocalAmountToDollars,
  convertToMaxSupportedPrecision,
} from 'src/localCurrency/convert'
import { fetchExchangeRate } from 'src/localCurrency/saga'
import { getLocalCurrencyCode, usdToLocalCurrencyRateSelector } from 'src/localCurrency/selectors'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import { UriData, uriDataFromUrl } from 'src/qrcode/schema'
import { AddressRecipient, Recipient, RecipientType } from 'src/recipients/recipient'
import { updateAppRecipientCache } from 'src/recipients/reducer'
import { TransactionDataInput } from 'src/send/types'
import { tokensListSelector } from 'src/tokens/selectors'
import { TokenBalance } from 'src/tokens/slice'
import { convertLocalToTokenAmount, getSupportedNetworkIdsForSend } from 'src/tokens/utils'
import { Currency } from 'src/utils/currencies'
import Logger from 'src/utils/Logger'
import { call, put, select } from 'typed-redux-saga'

const TAG = 'send/utils'

export function* handleSendPaymentData(
  data: UriData,
  isFromScan: boolean,
  cachedRecipient?: Recipient
) {
  const recipient: AddressRecipient = {
    address: data.address.toLowerCase(),
    name: data.displayName || cachedRecipient?.name,
    e164PhoneNumber: data.e164PhoneNumber,
    displayNumber: cachedRecipient?.displayNumber,
    thumbnailPath: cachedRecipient?.thumbnailPath,
    contactId: cachedRecipient?.contactId,
    recipientType: RecipientType.Address,
  }
  yield* put(
    updateAppRecipientCache({
      [data.address.toLowerCase()]: recipient,
    })
  )

  const supportedNetworkIds = yield* select(getSupportedNetworkIdsForSend)
  const tokens: TokenBalance[] = yield* select((state) =>
    tokensListSelector(state, supportedNetworkIds)
  )
  const tokenInfo = tokens.find((token) => token?.symbol === (data.token ?? Currency.Dollar))

  if (!tokenInfo?.priceUsd) {
    navigate(Screens.SendEnterAmount, {
      recipient,
      isFromScan,
      origin: SendOrigin.AppSendFlow,
      defaultTokenIdOverride: data.token ? tokenInfo?.tokenId : undefined,
      forceTokenId: !!(data.token && tokenInfo?.tokenId),
    })
    return
  }

  if (data.amount && tokenInfo?.address) {
    const currency: LocalCurrencyCode = data.currencyCode
      ? (data.currencyCode as LocalCurrencyCode)
      : yield* select(getLocalCurrencyCode)
    const exchangeRate = yield* call(fetchExchangeRate, currency)
    const dollarAmount = convertLocalAmountToDollars(data.amount, exchangeRate)
    const usdToLocalRate = yield* select(usdToLocalCurrencyRateSelector)
    const localAmount = convertDollarsToLocalAmount(dollarAmount, usdToLocalRate)
    const inputAmount = localAmount && convertToMaxSupportedPrecision(localAmount)
    const tokenAmount = convertLocalToTokenAmount({
      localAmount: inputAmount,
      tokenInfo,
      usdToLocalRate,
    })
    if (!inputAmount || !tokenAmount) {
      Logger.warn(TAG, '@handleSendPaymentData null amount')
      return
    }
    const transactionData: TransactionDataInput = {
      recipient,
      inputAmount,
      amountIsInLocalCurrency: true,
      tokenAddress: tokenInfo.address,
      tokenAmount,
      tokenId: tokenInfo.tokenId,
    }

    navigate(Screens.SendConfirmation, {
      transactionData,
      isFromScan,
      origin: SendOrigin.AppSendFlow,
    })
  } else {
    navigate(Screens.SendEnterAmount, {
      recipient,
      isFromScan,
      origin: SendOrigin.AppSendFlow,
      defaultTokenIdOverride: data.token ? tokenInfo?.tokenId : undefined,
      forceTokenId: !!(data.token && tokenInfo?.tokenId),
    })
  }
}

export function* handlePaymentDeeplink(deeplink: string) {
  try {
    const paymentData = uriDataFromUrl(deeplink)
    yield* call(handleSendPaymentData, paymentData, true)
  } catch (e) {
    Logger.warn('handlePaymentDeepLink', `deeplink ${deeplink} failed with ${e}`)
  }
}
