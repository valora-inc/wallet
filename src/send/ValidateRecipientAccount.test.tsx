import { fireEvent, render } from '@testing-library/react-native'
import * as React from 'react'
import { Provider } from 'react-redux'
import { SendOrigin } from 'src/analytics/types'
import { AddressValidationType } from 'src/identity/reducer'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import ValidateRecipientAccount from 'src/send/ValidateRecipientAccount'
import { createMockStore, getMockStackScreenProps } from 'test/utils'
import {
  mockAccountInvite,
  mockCusdAddress,
  mockE164NumberInvite,
  mockTransactionData,
} from 'test/values'

describe('ValidateRecipientAccount', () => {
  it('renders correctly when full validation required', () => {
    const store = createMockStore({
      identity: {
        secureSendPhoneNumberMapping: {
          [mockE164NumberInvite]: {
            addressValidationType: AddressValidationType.FULL,
            validationSuccessful: false,
          },
        },
      },
    })
    const tree = render(
      <Provider store={store}>
        <ValidateRecipientAccount
          {...getMockStackScreenProps(Screens.ValidateRecipientAccount, {
            transactionData: mockTransactionData,
            recipient: mockTransactionData.recipient,
            origin: SendOrigin.AppSendFlow,
          })}
        />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when partial validation required', () => {
    const store = createMockStore({
      identity: {
        secureSendPhoneNumberMapping: {
          [mockE164NumberInvite]: {
            addressValidationType: AddressValidationType.PARTIAL,
            validationSuccessful: false,
          },
        },
      },
    })
    const tree = render(
      <Provider store={store}>
        <ValidateRecipientAccount
          {...getMockStackScreenProps(Screens.ValidateRecipientAccount, {
            transactionData: mockTransactionData,
            recipient: mockTransactionData.recipient,
            origin: SendOrigin.AppSendFlow,
          })}
        />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  it('typing correct last four of account enables submit button', () => {
    const store = createMockStore({
      identity: {
        secureSendPhoneNumberMapping: {
          [mockE164NumberInvite]: {
            addressValidationType: AddressValidationType.PARTIAL,
            validationSuccessful: false,
          },
        },
      },
    })
    const tree = render(
      <Provider store={store}>
        <ValidateRecipientAccount
          {...getMockStackScreenProps(Screens.ValidateRecipientAccount, {
            transactionData: mockTransactionData,
            recipient: mockTransactionData.recipient,
            origin: SendOrigin.AppSendFlow,
          })}
        />
      </Provider>
    )
    expect(tree.getByTestId('ConfirmAccountButton')).toBeDisabled()
    fireEvent.changeText(tree.getByTestId('SingleDigitInput/digit0'), mockCusdAddress[38])
    fireEvent.changeText(tree.getByTestId('SingleDigitInput/digit1'), mockCusdAddress[39])
    fireEvent.changeText(tree.getByTestId('SingleDigitInput/digit2'), mockCusdAddress[40])
    fireEvent.changeText(tree.getByTestId('SingleDigitInput/digit3'), mockCusdAddress[41])
    expect(tree.getByTestId('ConfirmAccountButton')).not.toBeDisabled()
  })

  it('enables submit button for full validation', () => {
    const store = createMockStore({
      identity: {
        secureSendPhoneNumberMapping: {
          [mockE164NumberInvite]: {
            addressValidationType: AddressValidationType.FULL,
            validationSuccessful: false,
          },
        },
      },
    })
    const tree = render(
      <Provider store={store}>
        <ValidateRecipientAccount
          {...getMockStackScreenProps(Screens.ValidateRecipientAccount, {
            transactionData: mockTransactionData,
            recipient: mockTransactionData.recipient,
            origin: SendOrigin.AppSendFlow,
          })}
        />
      </Provider>
    )

    expect(tree.getByTestId('ConfirmAccountButton')).toBeDisabled()
    fireEvent.changeText(
      tree.getByTestId('ValidateRecipientAccount/TextInput'),
      mockCusdAddress[38]
    )
    expect(tree.getByTestId('ConfirmAccountButton')).not.toBeDisabled()
  })

  it('navigates to send confirmation when validation successful for send flow', () => {
    const store = createMockStore({
      identity: {
        secureSendPhoneNumberMapping: {
          [mockE164NumberInvite]: {
            addressValidationType: AddressValidationType.NONE,
            validationSuccessful: false,
          },
        },
      },
    })

    const props = getMockStackScreenProps(Screens.ValidateRecipientAccount, {
      transactionData: mockTransactionData,
      recipient: mockTransactionData.recipient,
      origin: SendOrigin.AppSendFlow,
    })

    const tree = render(
      <Provider store={store}>
        <ValidateRecipientAccount {...props} />
      </Provider>
    )

    const updatedStore = createMockStore({
      identity: {
        secureSendPhoneNumberMapping: {
          [mockE164NumberInvite]: {
            addressValidationType: AddressValidationType.NONE,
            address: mockAccountInvite,
            validationSuccessful: true,
          },
        },
      },
    })

    tree.rerender(
      <Provider store={updatedStore}>
        <ValidateRecipientAccount {...props} />
      </Provider>
    )

    expect(navigate).toHaveBeenCalledWith(Screens.SendConfirmation, {
      origin: SendOrigin.AppSendFlow,
      transactionData: mockTransactionData,
      isFromScan: false,
    })
  })
})
