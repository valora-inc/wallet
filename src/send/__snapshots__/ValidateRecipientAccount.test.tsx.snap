// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateRecipientAccount renders correctly when full validation required 1`] = `
[
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
      }
    }
    keyboardShouldPersistTaps="always"
    testID="KeyboardAwareScrollView"
  >
    <View>
      <View>
        <Text
          style={
            {
              "color": "#2E3338",
              "fontFamily": "Jost-Medium",
              "fontSize": 22,
              "lineHeight": 28,
              "paddingVertical": 16,
            }
          }
          testID="ConfirmAccountNumber/Title"
        >
          confirmAccountNumber.title
        </Text>
        <View>
          <View>
            <Text
              style={
                {
                  "color": "#2E3338",
                  "fontFamily": "Inter-Regular",
                  "fontSize": 16,
                  "lineHeight": 22,
                  "paddingBottom": 16,
                }
              }
            >
              confirmAccountNumber.body1Full, {"displayName":"Jane Doe"}
            </Text>
            <Text
              style={
                {
                  "color": "#2E3338",
                  "fontFamily": "Inter-Regular",
                  "fontSize": 16,
                  "lineHeight": 22,
                  "paddingBottom": 16,
                }
              }
            >
              confirmAccountNumber.body2Full
            </Text>
            <Text
              style={
                {
                  "color": "#2E3338",
                  "fontFamily": "Inter-SemiBold",
                  "fontSize": 14,
                  "lineHeight": 18,
                  "paddingVertical": 8,
                }
              }
            >
              accountInputHeaderFull
            </Text>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  {
                    "backgroundColor": "#FFF",
                    "borderColor": "#E6E6E6",
                    "borderRadius": 3,
                    "borderWidth": 1,
                    "flex": 0,
                    "height": 50,
                    "marginVertical": 5,
                  },
                ]
              }
            >
              <TextInput
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="0xf1b1d5a6e7728g309c4a025k122d71ad75a61976"
                shouldShowClipboard={[Function]}
                style={
                  [
                    {
                      "color": "#2E3338",
                      "flex": 1,
                      "fontFamily": "Inter-Regular",
                      "fontSize": 16,
                      "lineHeight": 22,
                      "paddingHorizontal": 0,
                      "paddingVertical": 12,
                    },
                    undefined,
                    undefined,
                  ]
                }
                testID="ValidateRecipientAccount/TextInput"
                value=""
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "height": 64,
            }
          }
        />
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "paddingVertical": 16,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 100,
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": true,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              focusable={false}
              nativeBackgroundAndroid={
                {
                  "attribute": "selectableItemBackground",
                  "rippleRadius": undefined,
                  "type": "ThemeAttrAndroid",
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#97DFC1",
                    "flexDirection": "row",
                    "height": 48,
                    "justifyContent": "center",
                    "minWidth": 120,
                    "opacity": undefined,
                    "paddingHorizontal": 24,
                    "paddingVertical": 5,
                  },
                  undefined,
                ]
              }
              testID="ConfirmAccountButton"
            >
              <Text
                maxFontSizeMultiplier={1}
                style={
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "Inter-SemiBold",
                    "fontSize": 16,
                    "lineHeight": 22,
                    "marginLeft": 0,
                    "marginRight": 0,
                  }
                }
              >
                confirmAccountNumber.submit
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <svg
          fill="none"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <circle
            cx="8"
            cy="8"
            r="7"
            stroke="#2E3338"
            strokeWidth="1.25"
          />
          <path
            d="M8 12V7M8 6V5"
            stroke="#2E3338"
            strokeWidth="1.25"
          />
        </svg>
        <Text
          onPress={[Function]}
          style={
            {
              "color": "#2E3338",
              "fontFamily": "Inter-Regular",
              "fontSize": 14,
              "lineHeight": 18,
              "paddingLeft": 8,
              "textDecorationLine": "underline",
            }
          }
        >
          confirmAccountNumber.help, {"displayName":"Jane Doe"}
        </Text>
      </View>
    </View>
  </RCTScrollView>,
  <View
    collapsable={false}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "right": 0,
        },
        {
          "height": 0,
        },
        undefined,
      ]
    }
  />,
  <Modal
    animationType="none"
    deviceHeight={1334}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={true}
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  />,
]
`;

exports[`ValidateRecipientAccount renders correctly when partial validation required 1`] = `
[
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
      }
    }
    keyboardShouldPersistTaps="always"
    testID="KeyboardAwareScrollView"
  >
    <View>
      <View>
        <Text
          style={
            {
              "color": "#2E3338",
              "fontFamily": "Jost-Medium",
              "fontSize": 22,
              "lineHeight": 28,
              "paddingVertical": 16,
            }
          }
          testID="ConfirmAccountNumber/Title"
        >
          confirmAccountNumber.title
        </Text>
        <View>
          <View>
            <Text
              style={
                {
                  "color": "#2E3338",
                  "fontFamily": "Inter-Regular",
                  "fontSize": 16,
                  "lineHeight": 22,
                  "paddingBottom": 16,
                }
              }
            >
              confirmAccountNumber.bodyPartial, {"displayName":"Jane Doe"}
            </Text>
            <Text
              style={
                {
                  "color": "#2E3338",
                  "fontFamily": "Inter-SemiBold",
                  "fontSize": 14,
                  "lineHeight": 18,
                  "paddingVertical": 8,
                }
              }
            >
              accountInputHeaderPartial
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  {
                    "paddingRight": 8,
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      {
                        "backgroundColor": "#FFF",
                        "borderColor": "#E6E6E6",
                        "borderRadius": 3,
                        "borderWidth": 1,
                        "flex": 0,
                        "height": 50,
                        "marginVertical": 5,
                        "width": 50,
                      },
                    ]
                  }
                >
                  <TextInput
                    maxLength={1}
                    multiline={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="a"
                    style={
                      [
                        {
                          "color": "#2E3338",
                          "flex": 1,
                          "fontFamily": "Inter-Regular",
                          "fontSize": 16,
                          "lineHeight": 22,
                          "paddingHorizontal": 0,
                          "paddingVertical": 12,
                        },
                        {
                          "textAlignVertical": "top",
                        },
                        undefined,
                      ]
                    }
                    testID="SingleDigitInput/digit0"
                    textAlign="center"
                    value=""
                  />
                </View>
              </View>
              <View
                style={
                  {
                    "paddingRight": 8,
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      {
                        "backgroundColor": "#FFF",
                        "borderColor": "#E6E6E6",
                        "borderRadius": 3,
                        "borderWidth": 1,
                        "flex": 0,
                        "height": 50,
                        "marginVertical": 5,
                        "width": 50,
                      },
                    ]
                  }
                >
                  <TextInput
                    maxLength={1}
                    multiline={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="0"
                    style={
                      [
                        {
                          "color": "#2E3338",
                          "flex": 1,
                          "fontFamily": "Inter-Regular",
                          "fontSize": 16,
                          "lineHeight": 22,
                          "paddingHorizontal": 0,
                          "paddingVertical": 12,
                        },
                        {
                          "textAlignVertical": "top",
                        },
                        undefined,
                      ]
                    }
                    testID="SingleDigitInput/digit1"
                    textAlign="center"
                    value=""
                  />
                </View>
              </View>
              <View
                style={
                  {
                    "paddingRight": 8,
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      {
                        "backgroundColor": "#FFF",
                        "borderColor": "#E6E6E6",
                        "borderRadius": 3,
                        "borderWidth": 1,
                        "flex": 0,
                        "height": 50,
                        "marginVertical": 5,
                        "width": 50,
                      },
                    ]
                  }
                >
                  <TextInput
                    maxLength={1}
                    multiline={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="F"
                    style={
                      [
                        {
                          "color": "#2E3338",
                          "flex": 1,
                          "fontFamily": "Inter-Regular",
                          "fontSize": 16,
                          "lineHeight": 22,
                          "paddingHorizontal": 0,
                          "paddingVertical": 12,
                        },
                        {
                          "textAlignVertical": "top",
                        },
                        undefined,
                      ]
                    }
                    testID="SingleDigitInput/digit2"
                    textAlign="center"
                    value=""
                  />
                </View>
              </View>
              <View
                style={
                  {
                    "paddingRight": 8,
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      {
                        "backgroundColor": "#FFF",
                        "borderColor": "#E6E6E6",
                        "borderRadius": 3,
                        "borderWidth": 1,
                        "flex": 0,
                        "height": 50,
                        "marginVertical": 5,
                        "width": 50,
                      },
                    ]
                  }
                >
                  <TextInput
                    maxLength={1}
                    multiline={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="4"
                    style={
                      [
                        {
                          "color": "#2E3338",
                          "flex": 1,
                          "fontFamily": "Inter-Regular",
                          "fontSize": 16,
                          "lineHeight": 22,
                          "paddingHorizontal": 0,
                          "paddingVertical": 12,
                        },
                        {
                          "textAlignVertical": "top",
                        },
                        undefined,
                      ]
                    }
                    testID="SingleDigitInput/digit3"
                    textAlign="center"
                    value=""
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "height": 64,
            }
          }
        />
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "paddingVertical": 16,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 100,
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": true,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              focusable={false}
              nativeBackgroundAndroid={
                {
                  "attribute": "selectableItemBackground",
                  "rippleRadius": undefined,
                  "type": "ThemeAttrAndroid",
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#97DFC1",
                    "flexDirection": "row",
                    "height": 48,
                    "justifyContent": "center",
                    "minWidth": 120,
                    "opacity": undefined,
                    "paddingHorizontal": 24,
                    "paddingVertical": 5,
                  },
                  undefined,
                ]
              }
              testID="ConfirmAccountButton"
            >
              <Text
                maxFontSizeMultiplier={1}
                style={
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "Inter-SemiBold",
                    "fontSize": 16,
                    "lineHeight": 22,
                    "marginLeft": 0,
                    "marginRight": 0,
                  }
                }
              >
                confirmAccountNumber.submit
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <svg
          fill="none"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <circle
            cx="8"
            cy="8"
            r="7"
            stroke="#2E3338"
            strokeWidth="1.25"
          />
          <path
            d="M8 12V7M8 6V5"
            stroke="#2E3338"
            strokeWidth="1.25"
          />
        </svg>
        <Text
          onPress={[Function]}
          style={
            {
              "color": "#2E3338",
              "fontFamily": "Inter-Regular",
              "fontSize": 14,
              "lineHeight": 18,
              "paddingLeft": 8,
              "textDecorationLine": "underline",
            }
          }
        >
          confirmAccountNumber.help, {"displayName":"Jane Doe"}
        </Text>
      </View>
    </View>
  </RCTScrollView>,
  <View
    collapsable={false}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "right": 0,
        },
        {
          "height": 0,
        },
        undefined,
      ]
    }
  />,
  <Modal
    animationType="none"
    deviceHeight={1334}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={true}
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  />,
]
`;
